From: Ana Guerrero Lopez <aguerrero@suse.com>
Date: Fri 21 Aug 2020 18:45:33 CEST
Subject: [PATCH] Replace hardcoded ../lib/charliecloud

All the resulting binaries and script will try to look for extra
scripts in /usr/lib/charliecloud while it should be /usr/lib64/charliecloud

diff -Nrua charliecloud-0.18/bin/ch-build charliecloud-0.18/bin/ch-build
--- charliecloud-0.18/bin/ch-build	2020-05-20 21:11:36.000000000 +0200
+++ charliecloud-0.18/bin/ch-build	2020-08-21 19:39:02.977368081 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -Nrua charliecloud-0.18/bin/ch-build2dir charliecloud-0.18/bin/ch-build2dir
--- charliecloud-0.18/bin/ch-build2dir	2020-05-20 21:11:36.000000000 +0200
+++ charliecloud-0.18/bin/ch-build2dir	2020-08-21 19:39:05.049368128 +0200
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -Nrua charliecloud-0.18/bin/ch-builder2squash charliecloud-0.18/bin/ch-builder2squash
--- charliecloud-0.18/bin/ch-builder2squash	2020-05-20 21:11:36.000000000 +0200
+++ charliecloud-0.18/bin/ch-builder2squash	2020-08-21 19:39:07.589368184 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -Nrua charliecloud-0.18/bin/ch-builder2tar charliecloud-0.18/bin/ch-builder2tar
--- charliecloud-0.18/bin/ch-builder2tar	2020-08-11 00:11:18.000000000 +0200
+++ charliecloud-0.18/bin/ch-builder2tar	2020-08-21 19:39:10.305368245 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -Nrua charliecloud-0.18/bin/ch-dir2squash charliecloud-0.18/bin/ch-dir2squash
--- charliecloud-0.18/bin/ch-dir2squash	2020-05-20 21:11:36.000000000 +0200
+++ charliecloud-0.18/bin/ch-dir2squash	2020-08-21 19:41:47.140625456 +0200
@@ -2,7 +2,7 @@
 
 set -e
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -Nrua charliecloud-0.18/bin/ch-fromhost charliecloud-0.18/bin/ch-fromhost
--- charliecloud-0.18/bin/ch-fromhost	2020-05-20 21:11:36.000000000 +0200
+++ charliecloud-0.18/bin/ch-fromhost	2020-08-21 19:39:16.765368390 +0200
@@ -35,7 +35,7 @@
 #   3. Here string, e.g. 'while IFS= read -r FILE; do ... done <<< "$FILES"'.
 #      This is a bashism.
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 set -e
diff -Nrua charliecloud-0.18/bin/ch-grow.py.in charliecloud-0.18/bin/ch-grow.py.in
--- charliecloud-0.18/bin/ch-grow.py.in	2020-08-19 01:04:31.000000000 +0200
+++ charliecloud-0.18/bin/ch-grow.py.in	2020-08-21 19:39:50.505369144 +0200
@@ -12,8 +12,7 @@
 import shutil
 import sys
 
-sys.path.insert(0, (  os.path.dirname(os.path.abspath(__file__))
-                    + "/../lib/charliecloud"))
+sys.path.insert(0, ("/usr/lib64/charliecloud"))
 import charliecloud as ch
 
 
diff -Nrua charliecloud-0.18/bin/ch-mount charliecloud-0.18/bin/ch-mount
--- charliecloud-0.18/bin/ch-mount	2020-05-20 21:11:36.000000000 +0200
+++ charliecloud-0.18/bin/ch-mount	2020-08-21 19:39:54.565369235 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -Nrua charliecloud-0.18/bin/ch-pull2dir charliecloud-0.18/bin/ch-pull2dir
--- charliecloud-0.18/bin/ch-pull2dir	2020-08-11 00:11:18.000000000 +0200
+++ charliecloud-0.18/bin/ch-pull2dir	2020-08-21 19:39:57.109369292 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 set -e
diff -Nrua charliecloud-0.18/bin/ch-pull2tar charliecloud-0.18/bin/ch-pull2tar
--- charliecloud-0.18/bin/ch-pull2tar	2020-05-20 21:11:36.000000000 +0200
+++ charliecloud-0.18/bin/ch-pull2tar	2020-08-21 19:39:59.781369352 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 set -e
diff -Nrua charliecloud-0.18/bin/ch-run-oci.py.in charliecloud-0.18/bin/ch-run-oci.py.in
--- charliecloud-0.18/bin/ch-run-oci.py.in	2020-08-19 01:04:31.000000000 +0200
+++ charliecloud-0.18/bin/ch-run-oci.py.in	2020-08-21 19:40:46.149370389 +0200
@@ -13,8 +13,7 @@
 import time
 import types
 
-sys.path.insert(0, (  os.path.dirname(os.path.abspath(__file__))
-                    + "/../lib/charliecloud"))
+sys.path.insert(0, ("/usr/lib64/charliecloud"))
 import version
 
 
diff -Nrua charliecloud-0.18/bin/ch-tar2dir charliecloud-0.18/bin/ch-tar2dir
--- charliecloud-0.18/bin/ch-tar2dir	2020-05-20 21:11:36.000000000 +0200
+++ charliecloud-0.18/bin/ch-tar2dir	2020-08-21 19:40:23.393369880 +0200
@@ -2,7 +2,7 @@
 
 set -e
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -Nrua charliecloud-0.18/bin/ch-test charliecloud-0.18/bin/ch-test
--- charliecloud-0.18/bin/ch-test	2020-08-11 00:11:18.000000000 +0200
+++ charliecloud-0.18/bin/ch-test	2020-08-21 19:40:28.321369990 +0200
@@ -20,7 +20,7 @@
 
 ### Setup
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 if [[ ! -f ${lib}/base.sh ]]; then
     fatal "install or build problem: not found: ${lib}/base.sh"
 fi
diff -Nrua charliecloud-0.18/bin/ch-tug.py.in charliecloud-0.18/bin/ch-tug.py.in
--- charliecloud-0.18/bin/ch-tug.py.in	2020-08-11 00:11:18.000000000 +0200
+++ charliecloud-0.18/bin/ch-tug.py.in	2020-08-21 19:40:54.201370569 +0200
@@ -4,8 +4,7 @@
 import os
 import sys
 
-sys.path.insert(0, (  os.path.dirname(os.path.abspath(__file__))
-                    + "/../lib/charliecloud"))
+sys.path.insert(0, ("/usr/lib64/charliecloud"))
 import charliecloud as ch
 
 
diff -Nrua charliecloud-0.18/bin/ch-umount charliecloud-0.18/bin/ch-umount
--- charliecloud-0.18/bin/ch-umount	2020-05-20 21:11:36.000000000 +0200
+++ charliecloud-0.18/bin/ch-umount	2020-08-21 19:40:33.345370103 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=/usr/lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -Nrua charliecloud-0.18.orig/lib/base.sh charliecloud-0.18/lib/base.sh
--- charliecloud-0.18/lib/base.sh	2020-08-11 00:11:18.000000000 +0200
+++ charliecloud-0.18/lib/base.sh	2020-08-21 19:58:14.470325969 +0200
@@ -6,7 +6,7 @@
 # shellcheck disable=SC2034
 ch_base=${ch_bin%/*}
 
-lib="${ch_bin}/../lib/charliecloud"
+lib="/usr/lib64/charliecloud"
 . "${lib}/version.sh"
 
 
